### PR TITLE
Add siac export command to support leaderboard

### PR DESCRIPTION
**Pull request by lukechampine**
This involved adding a field to the `api.RenterContract` struct. Since it's JSON, there shouldn't be any compatibility issues with the current frontends.

### CODE REVIEW

**Comment by lukechampine** [comment]
I also added a commit to omit empty transaction fields from the JSON. When exporting my 44 contracts, this reduces the filesize from 100KB to 70KB. It should also significantly reduce the amount of JSON returned by calls like `/wallet/transactions`.
Changing an encoding is always risky business, especially in the `types` package, but I think it's acceptable for JSON encodings. We never hash JSON, after all. However, it may be an issue for Sia-UI if there is code that explicitly checks for `null`, since now those fields will be `undefined` instead. cc @johnathanhowell

**Comment by lukechampine** [comment]
Reverted after further discussion. We can't break the API this way (the commit removes fields that were previously present in API responses).

**Comment by DavidVorick** [comment]
You may be able to get away with calling omitempty on the new field

**Comment by lukechampine** [comment]
true, but the new field isn't ever empty. The next step would be to convert the `types.Transaction` to a new api-specific type that omits empty fields, but that's just confusing.

**Comment by DavidVorick** [comment]
Probably best just not to use the tag anywhere in the API, bandwidth and spacing typically isn't the bottleneck anyway. A few hundred kb in a large request is okay.

File: api/consensus.go [hunk 1]
```diff
@@ -28,3 +29,20 @@ func (api *API) consensusHandler(w http.ResponseWriter, req *http.Request, _ htt
 		Target:       currentTarget,
 	})
 }
+
+// consensusValidateTransactionSetHandler handles the API calls to
+// /consensus/validate/transactionset.
+func (api *API) consensusValidateTransactionSetHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
```
Comment (DavidVorick) [inline api/consensus.go]: convention is to only captialize where the `/`s are

**Comment by lukechampine** [comment]
keep getting an appveyor error:
```
--- FAIL: TestIntegrationLoad1of1Siag (0.48s)
	unseeded_test.go:47: rename C:\Users\appveyor\AppData\Local\Temp\1\SiaTesting\wallet\TestIntegrationLoad1of1Siag\wallet\wallet.json_temp C:\Users\appveyor\AppData\Local\Temp\1\SiaTesting\wallet\TestIntegrationLoad1of1Siag\wallet\wallet.json: The process cannot access the file because it is being used by another process.
```

File: doc/API.md [hunk 2]
```diff
@@ -176,6 +177,19 @@ returns information about the consensus set, such as the current block height.
 }
 ```
 
+#### /consensus/validate/transactionset [POST]
+
+validates a set of transactions using the current utxo set.
```
Comment (DavidVorick) [inline doc/API.md]: Determines whether a transaction set is valid using the current state of the network. A valid transaction could be included in the next block if mined.

**Pull request merged**